src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -411,9 +411,21 @@ private TypeParameterConstraintClause GetDefaultTypeParameterConstraintClause(Ty
                 diagnostics.Add(ErrorCode.ERR_BadVisBound, location, containingSymbol, constraintType.Type);
             }
 
-            if (constraintType.Type.IsFileTypeOrUsesFileTypes() && !containingSymbol.ContainingType.IsFileTypeOrUsesFileTypes())
+            if (constraintType.Type.IsFileTypeOrUsesFileTypes() && (containingSymbol as TypeSymbol ?? containingSymbol.ContainingSymbol as TypeSymbol)?.IsFileTypeOrUsesFileTypes() == false)
             {
-                diagnostics.Add(ErrorCode.ERR_FileTypeDisallowedInSignature, location, constraintType.Type, containingSymbol);
+                // if the containing symbol of the constraint is a member, we need to ensure the nearest containing type is a file type.
+                var possibleFileType = containingSymbol switch
+                {
+                    TypeSymbol typeSymbol => typeSymbol,
+                    LocalFunctionSymbol => null,
+                    MethodSymbol method => (TypeSymbol)method.ContainingSymbol,
+                    _ => throw ExceptionUtilities.UnexpectedValue(containingSymbol)
+                };
+                Debug.Assert(possibleFileType?.IsDefinition != false);
+                if (possibleFileType?.IsFileTypeOrUsesFileTypes() == false)
```

Comment:
>if (possibleFileType?.IsFileTypeOrUsesFileTypes() == false) [](http://example.com/codeflow?start=16&length=59)

Is this check meant to replace the `&& (containingSymbol as TypeSymbol ?? ... )?.IsFileTypeOrUsesFileTypes() == false` above?

Reply:
Thanks, looks like I forgot to delete the original check.